uaclient/tests/test_util.py

```diff
@@ -295,22 +295,27 @@ def test_default_do_not_retry_on_failure_return_code(self, m_sleep, _subp):
             with pytest.raises(exceptions.ProcessExecutionError) as excinfo:
                 util.subp(["ls", "--bogus"])
 
-        expected_errors = [
-            "Failed running command 'ls --bogus' [exit(2)].",
-            "ls: unrecognized option '--bogus'",
-        ]
-        for msg in expected_errors:
-            assert msg in str(excinfo.value)
+        exec_class = excinfo._excinfo[1]
+        exit_code = exec_class.exit_code
+        out = exec_class.stdout
+        expected_exit_code = 2
+        assert exit_code == expected_exit_code
+        assert "" == out
```

Comment:
We can make it shorter:
```suggestion
        assert 2 == excinfo.value.exit_code
        assert "" == excinfo.value.stdout
```

Reply:
or check `out` from the subp call directly maybe